ILSpy.AddIn/Samples/ILSpyAddInSamples.cs

```diff
@@ -0,0 +1,420 @@
+		public struct NestedStruct : NestedInterface
+		{
+			// F:ILSpy.AddIn.Tests.SomeClass.NestedStruct.X
+			public int X;
+			// F:ILSpy.AddIn.Tests.SomeClass.NestedStruct.Y
+			public int Y;
+
+			// P:ILSpy.AddIn.Tests.SomeClass.NestedStruct.SomeProperty
+			public int SomeProperty
+			{
+				get { return X + Y; }
+			}
+
+			// M:ILSpy.AddIn.Tests.SomeClass.NestedStruct.SomeMethod
+			public int SomeMethod()
+			{
+				return X * Y;
+			}
+		}
+	}
+
+	// T:ILSpy.AddIn.Tests.SomeGenericClass`2
+	public class SomeGenericClass<T1, T2>
+	{
+		// F:ILSpy.AddIn.Tests.SomeGenericClass`2.mField1
+		T1 mField1;
+		// F:ILSpy.AddIn.Tests.SomeGenericClass`2.mField2
+		T2 mField2;
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.#ctor(`0,`1)
+		public SomeGenericClass(T1 a, T2 b)
+		{
+			mField1 = a;
+			mField2 = b;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.#ctor(`0)
+		public SomeGenericClass(T1 a)
+		{
+			mField1 = a;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.#ctor
+		public SomeGenericClass()
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.Finalize
+		~SomeGenericClass()
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassMethod(`0,`1)
+		public void GenericClassMethod(T1 a, T2 b)
+		{
+			mField1 = a;
+			mField2 = b;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassMethod(System.Int32*)
+		unsafe public void GenericClassMethod(int* x)
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassMethod(System.Int32*[])
+		unsafe public void GenericClassMethod(int*[] x)
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassMethod(`0[]@)
+		unsafe public void GenericClassMethod(out T1[] x)
+		{
+			x = null;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassMethod(System.Int32@)
+		public void GenericClassMethod(ref int a)
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassGenericMethod``1(``0@)
+		public T3 GenericClassGenericMethod<T3>(ref T3 x)
+		{
+			return x;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassGenericMethod``1
+		public T3 GenericClassGenericMethod<T3>()
+		{
+			return default(T3);
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassGenericMethod``1(System.Int32)
+		public void GenericClassGenericMethod<T3>(int x)
+		{
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassGenericMethod``2(`0,``1,System.Int32,`1@)
+		public T4 GenericClassGenericMethod<T3, T4>(T1 x, T4 y, int z, out T2 result)
+		{
+			mField1 = x;
+			string foo = y.ToString() + z.ToString();
+			result = mField2;
+			return y;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.GenericClassGenericMethod``2(`0,``0,System.Int32[])
+		public T3 GenericClassGenericMethod<T3, T4>(T1 x, T3 y, int[] z)
+		{
+			mField1 = x;
+			string foo = y.ToString() + z.ToString();
+			return y;
+		}
+
+		// M:ILSpy.AddIn.Tests.SomeGenericClass`2.ArrayMethod(`0[],System.Int32[])
+		public void ArrayMethod(T1[] x, int[] y)
```

Comment:
The first and last of these four ArrayMethod overloads are navigated to correctly, but the second and third incorrectly navigate to the fourth. Possible bug in /navigateTo implementation?